ckb/src/main/java/org/nervos/ckb/transaction/CellCollector.java

```diff
@@ -58,41 +58,42 @@ public CollectResult collectInputs(
             tx.outputsData,
             Collections.emptyList());
 
-    BigInteger inputsCapacity = BigInteger.ZERO;
+    long inputsCapacity = 0;
     for (CellInput cellInput : tx.inputs) {
       cellInputs.add(cellInput);
 
       CellWithStatus cellWithStatus = api.getLiveCell(cellInput.previousOutput, false);
-      inputsCapacity = inputsCapacity.add(cellWithStatus.cell.output.capacity);
+      initialLength += cellWithStatus.cell.output.capacity;
     }
     final List witnesses = new ArrayList<>();
 
     CellOutput changeOutput = tx.outputs.get(tx.outputs.size() - 1);
     boolean haveChangeOutput = false;
     //  If the last cellOutput's capacity is not zero,  it means there is no changeOutput
-    if (changeOutput.capacity.compareTo(BigInteger.ZERO) == 0) {
+    if (changeOutput.capacity == 0) {
       haveChangeOutput = true;
     }
 
-    BigInteger needCapacity = BigInteger.ZERO;
+    long needCapacity = 0;
     for (CellOutput cellOutput : tx.outputs) {
-      needCapacity = needCapacity.add(cellOutput.capacity);
+      needCapacity += cellOutput.capacity;
     }
 
     while (iterator.hasNext()) {
       TransactionInput transactionInput = iterator.next();
       if (transactionInput == null) break;
       CellInput cellInput = transactionInput.input;
-      inputsCapacity = inputsCapacity.add(transactionInput.capacity);
+//      inputsCapacity = inputsCapacity.add(transactionInput.capacity);
+      inputsCapacity += transactionInput.capacity;
       List<CellInput> cellInputList = lockInputsMap.get(transactionInput.lockHash);
       cellInputList.add(cellInput);
       cellInputs.add(cellInput);
       witnesses.add("0x");
       transaction.inputs = cellInputs;
       transaction.witnesses = witnesses;
-      BigInteger sumNeedCapacity =
+      long sumNeedCapacity =
           calSumNeedCapacity(feeRate, transaction, changeOutput, haveChangeOutput, needCapacity);
-      if (inputsCapacity.compareTo(sumNeedCapacity) > 0) {
+      if (inputsCapacity > sumNeedCapacity) {
```

Comment:
Please use [Long.compareUnsigned](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Long.html#compareUnsigned(long,long))

Reply:
I mention this in the PR description

> I didn't use unsigned method to manipulate capacity as they won't exceed LONG.MAX_VALUE in real cases and can work correctly with signed operations.